nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenStat.scala

```diff
@@ -588,6 +591,52 @@ trait NirGenStat[G <: nsc.Global with Singleton] { self: NirGenPhase[G] =>
           ()
       }
 
+    def genJavaDefaultMethodBody(dd: DefDef): Seq[nir.Inst] = {
+      val fresh = Fresh()
+      val buf   = new ExprBuffer()(fresh)
+
+      implicit val pos: nir.Position = dd.pos
+
+      val sym                    = dd.symbol
+      val forwardeeClassFullName = sym.owner.fullName + "$class"
+
+      val forwardeeClassSym = findMemberFromRoot(
+        TermName(forwardeeClassFullName))
+
+      val forwardeeMethodSym = forwardeeClassSym.info
+        .member(sym.name)
+        .suchThat { s =>
+          s.isMethod &&
+          s.tpe.params.size == sym.tpe.params.size + 1 &&
+          s.tpe.params.head.tpe =:= sym.owner.toTypeConstructor &&
+          s.tpe.params.tail.zip(sym.tpe.params).forall {
+            case (sParam, symParam) =>
+              sParam.tpe =:= symParam.tpe
+          }
+        }
+
+      val forwardeeName =
+        Val.Global(genMethodName(forwardeeMethodSym), Type.Ptr)
+      val forwardeeSig = genMethodSig(forwardeeMethodSym)
+
+      val Type.Function(paramtys, retty) = forwardeeSig
+
+      val params = paramtys.map(ty => Val.Local(fresh(), ty))
+      buf.label(fresh(), params)
+
+// Reviewer(s): The genFuncPtrExternForwarder() code boxes the parameters.
+//           Does that need to be done here, since this is calling
+//           Scala code? Since it works, I think not, but
+//           appreciate confirmation. "Think not" is not good enough for
+//           release code.
+
+      val res        = buf.call(forwardeeSig, forwardeeName, params, Next.None)
+      val unboxedRes = buf.toExtern(retty, res)
+      buf.ret(unboxedRes)
```

Comment:
We should not unbox the result either. The result type is exactly the same.
```suggestion
      buf.ret(res)
```